### ISSUE TITLE
Links to local files don’t work

### ISSUE THREAD

**Issue by wooorm** [IBM/nicedoc.io#40]
(or don’t always work)

Take the readme of franc, https://nicedoc.io/wooorm/franc, at the bottom it links to the `LICENSE` file.
nicedoc generates a link with `href="LICENSE"` from that, which on nicedoc goes to `https://nicedoc.io/wooorm/LICENSE` (a 500 which should be a 404?)

Also thanks for using remark and unified!! 🤗

**Comment by Kikobeats** [comment]
An interesting thing is happening here.

The original markdown markup doesn't have a link to somewhere

```markdown
## License

[MIT][] © [Titus Wormer][home]
```

so when the markdown is converted into html, there is not too much to do:

```html
<h2 id="license">License</h2>
<p><a href="LICENSE">MIT</a> © <a href="http://wooorm.com">Titus Wormer</a></p>
```

Ideally, this should be the kinds of things where nicedoc is powerful: for normalizing things.

But in those cases, I feel we can't do too much since it's not present as part of the input.

Thoughts?

**Comment by wooorm** [comment]
That’s not the whole truth: what’s happening with `[mit][]` (and later `[mit]: license`), is called a [link reference](https://spec.commonmark.org/0.29/#reference-link) (and a [link definition](https://spec.commonmark.org/0.29/#link-reference-definitions)).
It’s handled the same as `[mit](license)` (and remark handles that correctly).

However, GitHub treats these links as going to the file (`foo/bar/blob/master/license`), but nicedoc.io doesn’t, and this issue is about that!

**Comment by wooorm** [comment]
(some more info, there’s shortcut references, `[foo]`, collapsed references, `[foo][]`, and full references, `[bar][foo]`)

**Comment by Kikobeats** [comment]
oh, I see. What do you think is the best way to handle that? Should we write a unified plugin? 🤔

**Comment by wooorm** [comment]
yeah! Something that checks all `.url`s in mdast, or `src/href/etc` in hast, and rewrites them? Not entirely sure what the algorithm should be for which cases to rewrite URLs though!

**Comment by wooorm** [comment]
Ah, something along [these lines](https://github.com/remarkjs/remark-validate-links/blob/caa1f9fa8ca09567951cf28f9252f642ab36ad9a/index.js#L274)

**Comment by Kikobeats** [comment]
Totally appreciated your help with that, since I'm a unified AST n00b 😆

**Comment by jjb** [comment]
Here's a possible interesting example:

https://github.com/RomuloOliveira/commit-messages-guide

https://nicedoc.io/RomuloOliveira/commit-messages-guide

try the language translation links. i'm not sure what expected behavior here is, but since it didn't just go to the github.com link, i'm assuming that nicedoc is trying to nicedocify it properly.
